Changes for jira pubstwo-1661:

orosz-usgs opened the pull request:
Made IpdsParserService.formatOrcid statci and public
  Updated IpdsParserService.formatOrcid to accept oricid using https
  Added nethod MvcService.normalizeFilters, to uses formatOrcid to change orcid value sent
from client to standard format
  Updated MvcService to normal data with normalizeFilters
  Updated IpdsParserServiceTest.testFormatOrcid to test for the https form of orcid

Commit:
Changes for jira pubstwo-1661:
  Made IpdsParserService.formatOrcid statci and public
  Updated IpdsParserService.formatOrcid to accept oricid using https
  Added nethod MvcService.normalizeFilters, to uses formatOrcid to change orcid value sent
from client to standard format
  Updated MvcService to normal data with normalizeFilters
  Updated IpdsParserServiceTest.testFormatOrcid to test for the https form of orcid

dsteinich commented on src/main/java/gov/usgs/cida/pubs/busservice/ipds/IpdsParserService.java:
```diff
@@ -57,15 +59,17 @@ protected Integer getFirstNodeInteger(final Element element, final String tagNam
 	}
 
-	protected String formatOrcid(String orcid) {
+	public static String formatOrcid(String orcid) {
```
Our respective code changes are are going to collide here. The entire *Ipds* related code is being removed. I like the idea of a static function to format the orcid - perhaps PubsUtils would be a good place to put it. I think we also talked about just storing the unique portion of the orcid, rather than the entire url. I was going to use something like:

Pattern ORCID_PATTERN = Pattern.compile("\\d{4}-\\d{4}-\\d{4}-(\\d{3}X|\\d{4})");
Matcher matcher = ORCID_PATTERN.matcher(orcid);
if (matcher.find()) {
	formattedOrcid = matcher.group();
} else {
	formattedOrcid = null;
}

dsteinich reviewed (changes requested):
The Travis CI test failure (DisseminationListServiceIT.processDisseminationListTest) is a brittle test that I am fixing as part of PUBSTWO-1647.

Commit:
Changes for jira pubstwo-1661:
Updates to support storing the short form of the ORCID in the database.
  Reverted changes to: IpdsParserService & IpdsParserServiceTest.java
  New method PubsUtils.normalizeOrcid to replaced IpdsParserService.formatOrcid, refactored as
suggested in the pull review. Updated to return short concanolical form that will be stored in
database.

Commit:
Merge branch 'master' of https://github.com/usgs/pubs-services into jira-pubstwo-1661

Commit:
Changes for jira pubstwo-1661:
  Fixed codacy issue, moved declartion of ORCID_PATTERN to first statement in the class
